app/controllers/accounts_controller.rb

```diff
@@ -0,0 +1,50 @@
+class AccountsController < ApplicationController
+  before_action :authenticate_user!
+  before_action :set_account, only: [:show, :edit, :update, :destroy]
+
+  def index
+    @accounts = current_user.accounts
+  end
+
+  def show
+  end
+
+  def new
+    @account = current_user.accounts.new
+  end
+
+  def create
+    @account = current_user.accounts.build(account_params)
+    if @account.save
+      redirect_to accounts_path, notice: 'Account was successfully created.'
+    else
+      render :new
+    end
+  end
+
+  def edit
+  end
+
+  def update
+    if @account.update(account_params)
+      redirect_to accounts_path, notice: 'Account was successfully updated.'
+    else
+      render :edit
+    end
+  end
+
+def destroy
+  @account.destroy
+  redirect_to accounts_path, notice: 'Account was successfully deleted.'
+end
+
+  private
+
+  def set_account
+    @account = current_user.accounts.find(params[:id])
```

Comment:
- What will happen if account is not found?
- How are you handling the above situation?